packages/kbn-esql-ast/src/ast_walker.ts

```diff
@@ -323,16 +323,21 @@ function getConstant(ctx: ConstantContext): ESQLAstItem {
     // e.g. 1 year, 15 months
     return createTimeUnit(ctx);
   }
+
+  // Decimal type covers multiple ES|QL types: long, double, etc.
   if (ctx instanceof DecimalLiteralContext) {
-    return createNumericLiteral(ctx.decimalValue());
+    return createNumericLiteral(ctx.decimalValue(), 'decimal');
```

Comment:
Initially, I expanded to DoubleLiteralContext, LongLiteralContext, FloatLiteralContext, etc, but since we need a single literal type to pass as multiple function parameter types (e.g. `1.` should work as a double or a long or float), so I condensed into its own Decimal literal type here.